regression/Makefile

```diff
@@ -20,18 +20,51 @@ DIRS = ansi-c \
        test-script \
        # Empty last line
 
-# Check for the existence of $dir. Tests under goto-gcc cannot be run on
-# Windows, so appveyor.yml unlinks the entire directory under Windows.
+# Run all test directories in sequence
 test:
 	@for dir in $(DIRS); do \
-	  if [ -d "$$dir" ]; then \
-	    $(MAKE) -C "$$dir" test || exit 1; \
-	  fi; \
+        $(MAKE) "test-$$dir" || exit 1; \
 	done;
 
+# Pattern to execute a single directory, checking for existence of the directory
+# first. Tests under goto-gcc cannot be run on Windows, so appveyor.yml unlinks
+# the entire directory under Windows.
+test-%:
+	@if [ -d "$*" ]; then \
+		$(MAKE) -C "$*" test || exit 1; \
+	fi;
+
+
+# Run all test directories in batch mode, enabling parallel execution of
+# test directories while avoiding interleaving of output, at the expense
+# of less interactive feedback.
+BATCH_TARGETS = $(patsubst %,.%-batch-output, $(DIRS))
+
+#.PHONY: test-batch $(BATCH_TARGETS)
+
+test-batch: $(BATCH_TARGETS)
+	@for batch in "$<" ; do \
+		if [ -f "$$batch" ]; then \
+			cat "$$batch"; \
+			rm "$$batch"; \
+		fi; \
+	done;
+
+# Run a single test directory, capturing it's stdout/stderr to
+# a file
+.%-batch-output:
+	@if [ -d "$*" ]; then \
+		echo "Running $*..." ; \
+		$(MAKE) "test-$*" > "$@" 2>&1 || (echo "Failed $* at $$(date)" ; cat "$@" ; exit 1) ; \
```

Comment:
No output if the tests all pass right? Seems reasonable mostly just noting that is a change in behaviour.

Reply:
Currently the `test-batch` rule will output all the output if no tests fail, so you'll still get the full output, just batched up to the end of test runs. This is all still a work in progress though, which is why I have the 'DO NOT MERGE' label on this PR at the moment. I want to play with a few different strategies first rather than knee jerk something in :-)